tests/backend/test_multimode_state_preparations.py

```diff
@@ -433,3 +433,193 @@ def test_multimode_gaussian_random_state(
+        assert np.allclose(state.covs(), np.array([cov]) * hbar / 2, atol=tol, rtol=0)
+
+        # test that displaced squeezed states are unchanged
+        ex_means, ex_V = displaced_squeezed_state(
+            np.abs(a), np.angle(a), r, phi, basis="gaussian", hbar=hbar
+        )
+        for i in [0, 2, 3]:
+            state = backend.state([i])
+            assert np.allclose(state.means(), np.array([ex_means]), atol=tol, rtol=0)
+            assert np.allclose(state.covs(), np.array([ex_V]), atol=tol, rtol=0)
+
+    def test_multimode_gaussian_state(self, setup_backend, batch_size, pure, tol, hbar):
+        """Test multimode Gaussian state preparation"""
+        N = 4
+        backend = setup_backend(N)
+
+        cov = np.diag(np.exp(2 * np.array([-1, -1, 1, 1])))
+        means = np.zeros([4])
+
+        # prepare displaced squeezed states in all modes
+        a = 0.2 + 0.4j
+        r = 0.5
+        phi = 0.12
+        for i in range(N):
+            backend.prepare_displaced_squeezed_state(np.abs(a), np.angle(a), r, phi, i)
+
+        # prepare new squeezed displaced state in mode 1 and 3
+        backend.prepare_gaussian_state(means, cov, modes=[1, 3])
+        state = backend.state([1, 3])
+
+        # test Gaussian state is correct
+        state = backend.state([1, 3])
+        assert np.allclose(
+            state.means(), np.array([means[from_xp(2)]]) * np.sqrt(hbar / 2), atol=tol, rtol=0
+        )
+        assert np.allclose(
+            state.covs(), np.array([cov[from_xp(2), :][:, from_xp(2)]]) * hbar / 2, atol=tol, rtol=0
+        )
+
+        # test that displaced squeezed states are unchanged
+        ex_means, ex_V = displaced_squeezed_state(
+            np.abs(a), np.angle(a), r, phi, basis="gaussian", hbar=hbar
+        )
+        for i in [0, 2]:
+            state = backend.state([i])
+            assert np.allclose(state.means(), np.array([ex_means[from_xp(1)]]), atol=tol, rtol=0)
+            assert np.allclose(
+                state.covs(), np.array([ex_V[from_xp(1), :][:, from_xp(1)]]), atol=tol, rtol=0
+            )
+
+    def test_full_mode_squeezed_state(self, setup_backend, batch_size, pure, tol, hbar):
+        """Test full register Gaussian state preparation"""
+        N = 4
+        backend = setup_backend(N)
+
+        cov = np.diag(np.exp(2 * np.array([-1, -1, -1, -1, 1, 1, 1, 1])))
+        means = np.zeros([8])
+
+        backend.reset(pure=pure)
+        backend.prepare_gaussian_state(means, cov, modes=range(N))
+        state = backend.state()
+
+        # test Gaussian state is correct
+        state = backend.state()
+        assert np.allclose(
+            state.means(), np.array([means[from_xp(4)]]) * np.sqrt(hbar / 2), atol=tol, rtol=0
+        )
+        assert np.allclose(
+            state.covs(), np.array([cov[from_xp(4), :][:, from_xp(4)]]) * hbar / 2, atol=tol, rtol=0
+        )
+
+    def test_multimode_gaussian_random_state(self, setup_backend, batch_size, pure, tol, hbar):
```

Comment:
As I said for this entire class of tests, it is a direct copy of the test for the Gaussian backend, the only change being that the basis ordering is different, and that BaseBosonicState has the method .covs instead of Gaussian .cov. If we change this one, we would also need to change the Gaussian backend test, which has been just fine.

Reply:
Done. The tests are now split, at the expense of some lines repeated lines.